### ISSUE TITLE
ModuleNotFoundError: No module named 'jsonschema'

### ISSUE THREAD

**Issue by SauravMaheshkar** [wandb/weave#3027]
With `weave==0.51.20` + `python v3.10.15`, there seems to be an import error:

```console
~/dev
➜ mkdir tmp

~/dev
➜ cd tmp

~/dev/tmp
➜ uv venv
Using CPython 3.10.15
Creating virtual environment at: .venv
Activate with: source .venv/bin/activate

~/dev/tmp
➜ source .venv/bin/activate

~/dev/tmp via 🐍 v3.10.15 (tmp)
➜ uv pip install -U weave
Resolved 49 packages in 128ms
Prepared 49 packages in 2ms
Installed 49 packages in 103ms
 + aiohappyeyeballs==2.4.3
 + aiohttp==3.11.6
 + aiosignal==1.3.1
 + annotated-types==0.7.0
 + anyio==4.6.2.post1
 + async-timeout==5.0.1
 + attrs==24.2.0
 + backoff==2.2.1
 + certifi==2024.8.30
 + charset-normalizer==3.4.0
 + click==8.1.7
 + docker-pycreds==0.4.0
 + emoji==2.14.0
 + exceptiongroup==1.2.2
 + frozenlist==1.5.0
 + gitdb==4.0.11
 + gitpython==3.1.43
 + gql==3.5.0
 + graphql-core==3.2.5
 + idna==3.10
 + markdown-it-py==3.0.0
 + mdurl==0.1.2
 + multidict==6.1.0
 + numpy==2.1.3
 + packaging==24.2
 + platformdirs==4.3.6
 + propcache==0.2.0
 + protobuf==5.28.3
 + psutil==6.1.0
 + pydantic==2.9.2
 + pydantic-core==2.23.4
 + pygments==2.18.0
 + pyyaml==6.0.2
 + requests==2.32.3
 + requests-toolbelt==1.0.0
 + rich==13.9.4
 + sentry-sdk==2.18.0
 + setproctitle==1.3.4
 + setuptools==75.5.0
 + six==1.16.0
 + smmap==5.0.1
 + sniffio==1.3.1
 + tenacity==9.0.0
 + typing-extensions==4.12.2
 + urllib3==2.2.3
 + uuid-utils==0.9.0
 + wandb==0.18.7
 + weave==0.51.20
 + yarl==1.17.2

~/dev/tmp via 🐍 v3.10.15 (tmp)
➜ python
Python 3.10.15 (main, Sep  9 2024, 22:43:48) [Clang 18.1.8 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import weave
```

<details>
<summary>Error</summary>

```console
>>> import weave
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/sauravmaheshkar/dev/tmp/.venv/lib/python3.10/site-packages/weave/__init__.py", line 4, in <module>
    from weave.trace.api import *
  File "/Users/sauravmaheshkar/dev/tmp/.venv/lib/python3.10/site-packages/weave/trace/api.py", line 15, in <module>
    from weave.trace import urls, util, weave_client, weave_init
  File "/Users/sauravmaheshkar/dev/tmp/.venv/lib/python3.10/site-packages/weave/trace/weave_client.py", line 22, in <module>
    from weave.trace.feedback import FeedbackQuery, RefFeedbackQuery
  File "/Users/sauravmaheshkar/dev/tmp/.venv/lib/python3.10/site-packages/weave/trace/feedback.py", line 16, in <module>
    from weave.trace.rich.refs import Refs
  File "/Users/sauravmaheshkar/dev/tmp/.venv/lib/python3.10/site-packages/weave/trace/rich/refs.py", line 11, in <module>
    from weave.trace.vals import WeaveObject
  File "/Users/sauravmaheshkar/dev/tmp/.venv/lib/python3.10/site-packages/weave/trace/vals.py", line 28, in <module>
    from weave.trace.serialize import from_json
  File "/Users/sauravmaheshkar/dev/tmp/.venv/lib/python3.10/site-packages/weave/trace/serialize.py", line 12, in <module>
    from weave.trace_server.interface.base_object_classes.base_object_registry import (
  File "/Users/sauravmaheshkar/dev/tmp/.venv/lib/python3.10/site-packages/weave/trace_server/interface/base_object_classes/base_object_registry.py", line 1, in <module>
    from weave.trace_server.interface.base_object_classes.actions import ActionSpec
  File "/Users/sauravmaheshkar/dev/tmp/.venv/lib/python3.10/site-packages/weave/trace_server/interface/base_object_classes/actions.py", line 3, in <module>
    import jsonschema
ModuleNotFoundError: No module named 'jsonschema'
```

</details>

**Comment by tssweeney** [comment]
Hello @SauravMaheshkar thank you for the bug report and apologies for the inconvenience. You are indeed correct that we missed a dependency. I have corrected the issue, released `0.51.21` with the fix, and yanked `0.51.20` from pip. Please update `weave` to fix the issue.

**Comment by SauravMaheshkar** [comment]
Thanks that fixed the issue, closing the issue as resolved.
